src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -189,32 +209,32 @@
   ([database :- i/DatabaseInstance db-metadata]
    ;; determine what's changed between what info we have and what's in the DB
    (let [db-tables               (table-set db-metadata)
-         our-metadata            (our-metadata database)
-         strip-desc              (fn [metadata]
-                                   (set (map #(dissoc % :description) metadata)))
+         name+schema->db-table   (m/index-by (juxt :name :schema) db-tables)
+         our-metadata            (db->our-metadata database)
+         keep-table-id-info      (fn [metadata]
+                                   (set (map #(select-keys % [:schema :name]) metadata)))
          [new-tables old-tables] (data/diff
-                                  (strip-desc db-tables)
-                                  (strip-desc our-metadata))
-         [changed-tables]        (data/diff db-tables our-metadata)]
+                                  (keep-table-id-info db-tables)
+                                  (keep-table-id-info our-metadata))]
      ;; update database metadata from database
      (when (some? (:version db-metadata))
        (sync-util/with-error-handling (format "Error creating/reactivating tables for %s"
                                               (sync-util/name-for-logging database))
          (update-database-metadata! database db-metadata)))
      ;; create new tables as needed or mark them as active again
      (when (seq new-tables)
-       (sync-util/with-error-handling (format "Error creating/reactivating tables for %s"
-                                              (sync-util/name-for-logging database))
-         (create-or-reactivate-tables! database new-tables)))
+       (let [new-tables-info (set (map #(get name+schema->db-table [(:name %) (:schema %)]) new-tables))]
+         (sync-util/with-error-handling (format "Error creating/reactivating tables for %s"
+                                                (sync-util/name-for-logging database))
+           (create-or-reactivate-tables! database new-tables-info))))
      ;; mark old tables as inactive
      (when (seq old-tables)
        (sync-util/with-error-handling (format "Error retiring tables for %s" (sync-util/name-for-logging database))
          (retire-tables! database old-tables)))
 
-     ;; update description for changed tables
-     (when (seq changed-tables)
-       (sync-util/with-error-handling (format "Error updating table description for %s" (sync-util/name-for-logging database))
-         (update-table-description! database changed-tables)))
+     (sync-util/with-error-handling (format "Error updating table metadata for %s" (sync-util/name-for-logging database))
```

Comment:
previously, only table description could be updated during sync step, but I've changed it so that it could extend to other properties too.